server/sonar-server/src/main/java/org/sonar/server/db/migrations/v51/CopyScmAccountsFromAuthorsToUsers.java

```diff
@@ -130,19 +131,12 @@ public void handle(Select.Row row) throws SQLException {
     return users;
   }
 
-  private static String convertScmAccountsToCsv(List<String> scmAccounts) {
-    List<String> list = newArrayList(scmAccounts);
-    // Add one empty character at the begin and at the end of the list to be able to generate a coma at the begin and at the end of the
-    // text
-    list.add(0, "");
-    list.add("");
-    int size = list.size();
-    StringWriter writer = new StringWriter(size);
-    CsvWriter csv = CsvWriter.of(writer);
-    csv.values(list.toArray(new String[size]));
-    csv.close();
-    // Remove useless line break added by CsvWriter at this end of the text
-    return CharMatcher.JAVA_ISO_CONTROL.removeFrom(writer.toString());
+  @CheckForNull
+  private static String encodeScmAccounts(List<String> scmAccounts) {
+    if (scmAccounts.isEmpty()) {
+      return null;
+    }
+    return UserDto.SCM_ACCOUNTS_SEPARATOR + Joiner.on(UserDto.SCM_ACCOUNTS_SEPARATOR).join(scmAccounts) + UserDto.SCM_ACCOUNTS_SEPARATOR;
```

Comment:
Are you sure it's a good idea to use a constant from UserDto ? If tomorrow we change the format and we update this constant, this migration won't work anymore.